manager/src/main/java/org/openremote/manager/security/ManagerKeycloakIdentityProvider.java

```diff
@@ -1021,13 +1021,92 @@ public ClientRepresentation createUpdateClient(String realm, ClientRepresentatio
+                    }
+                }
+            }
+
+            return identityProvidersResource.get(alias).toRepresentation();
+        });
+    }
+
+    public void createUpdateIdentityProviderMapper(String realm, String idpAlias, String mapperName, String idpMapper, Map<String, String> mapperConfig) {
+        getRealms(realmsResource -> {
+            IdentityProviderResource idpResource = realmsResource.realm(realm).identityProviders().get(idpAlias);
+
+            IdentityProviderMapperRepresentation mapper = new IdentityProviderMapperRepresentation();
+            mapper.setName(mapperName);
+            mapper.setConfig(mapperConfig);
+            mapper.setIdentityProviderAlias(idpAlias);
+            mapper.setIdentityProviderMapper(idpMapper);
+
+            Optional<IdentityProviderMapperRepresentation> existingMapper = idpResource.getMappers().stream().filter(m -> mapperName.equals(m.getName())).findFirst();
+            if (existingMapper.isPresent()) {
+                idpResource.update(existingMapper.get().getId(), mapper);
+            } else {
+                try (Response response = idpResource.addMapper(mapper)) {
+                    if (response.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
+                        LOG.warning("Failed to create identity provider mapper '" + mapperName +  "' response=" + response.getStatusInfo().getStatusCode() + ": " + response.getStatusInfo().getReasonPhrase());
+                    }
+                }
+            }
+
+            return null;
+        });
+    }
+
+    public void addAuthenticationExecutionConfig(String realm, String flowAlias, String executionProviderId, String authenticatorAlias, Map<String, String> authenticatorConfig) {
+        getRealms(realmsResource -> {
+            AuthenticationManagementResource authenticationManagementResource = realmsResource.realm(realm).flows();
+            List<AuthenticationExecutionInfoRepresentation> executions = authenticationManagementResource.getExecutions(flowAlias);
+            for (AuthenticationExecutionInfoRepresentation execution : executions) {
+                if (executionProviderId.equals(execution.getProviderId())) {
+                    AuthenticatorConfigRepresentation config = new AuthenticatorConfigRepresentation();
+                    config.setAlias(authenticatorAlias);
+                    config.setConfig(authenticatorConfig);
+
+                    try (Response response = authenticationManagementResource.newExecutionConfig(execution.getId(), config)) {
+                        if (response.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
+                            LOG.warning("Failed to create execution config '" + authenticatorAlias + "' response=" + response.getStatusInfo().getStatusCode() + ": " + response.getStatusInfo().getReasonPhrase());
+                        }
+                    }
+                    return null;
```

Comment:
Shouldn't there be a `catch` statement here?

Reply:
It's a [try-with-resouces statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) to make sure the resource is automatically closed.